### DISCUSSION TITLE
Testing Actions locally

### DISCUSSION THREAD

**Discussion by BenjaminSSL** [ory/.github#86] [Q&A] [answered]
I'm developing a backend services, which should handle webhooks coming from Ory Actions.

Im using the managed version of ORY. When i try to set the url in an action to localhost or 127.0.0.1, it just gives me an error.

I wanna be able to test the webhooks on my local machine before uploading my code to production. How to do this?

My general idea is that I need to use a tunnel or proxy, however I cannot figure out how to.

**Comment by BenjaminSSL** [comment, marked as answer]
I have found working solution. At the moment im running a prod and dev project. For the dev environment I have setup a cloudflare tunnel pointing to a localhost connector.

This works fine for now. However it would be nice if I did not have to use a cloudflare tunnel.

**Reply by aeneasr** [reply]
I agree with this! We have it on the backlog but are currently busy with other topics. Great that you have been able to use cloudflare tunnel to address this! :)
